### PR TITLE
✨ Add support for uploading assets

### PR DESCRIPTION
**Pull request by laurentsimon**
Upload the provenance and the binary as assets to the release, when the workflow runs on a push with new tag.
closes https://github.com/slsa-framework/slsa-github-generator-go/issues/58

### CODE REVIEW

File: .github/workflows/slsa3_builder.yml [hunk 1]
```diff
@@ -410,3 +418,60 @@ jobs:
...
+#                                                                 #
+#           Upload binaries and provenances as assets             #
+#                                                                 #
+###################################################################
+  upload-assets:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    needs: [build,build-dry,provenance]
+    if: startsWith(github.ref, 'refs/tags/') && ${{ inputs.upload-assets == 'true' }}
```
Comment (laurentsimon) [inline .github/workflows/slsa3_builder.yml:432]: this line is not working. I'll troubleshoot that. But the rest is ready for review.

Comment (laurentsimon) [reply .github/workflows/slsa3_builder.yml]: Fixed.

File: .github/workflows/slsa3_builder.yml [hunk 2]
```diff
@@ -410,3 +418,60 @@ jobs:
...
+          echo "hash of provenance $UNTRUSTED_PROVENANCE_NAME should be $UNTRUSTED_PROVENANCE_HASH"
+
+          COMPUTED_HASH=$(sha256sum "$UNTRUSTED_PROVENANCE_NAME" | awk '{print $1}')
+          echo "provenance hash computed is $COMPUTED_HASH"
+
+          # Compare hashes. Explicit exit to be safe.
+          echo "$UNTRUSTED_PROVENANCE_HASH $UNTRUSTED_PROVENANCE_NAME" | sha256sum --strict --check --status || exit -2
+
+      - name: Release
+        uses: softprops/action-gh-release@1e07f4398721186383de40550babbdf2b84acfc5
```
Comment (laurentsimon) [inline .github/workflows/slsa3_builder.yml:473]: Note: we could later replace this with our own (it's just 2 API calls) if we don't want to rely on a 3P action to do it (it needs `contents:write`)

Comment (joshuagl) [reply .github/workflows/slsa3_builder.yml:473]: In general, I prefer not to rely on third-party actions and as a third-party action ourselves I think it makes a lot of sense to constrain the amount of parties we require users to trust. Let's replace this later.

**Review by joshuagl — APPROVED** [review]
Very cool!

File: .github/workflows/slsa3_builder.yml (hunk 2 above)
Comment (laurentsimon) [reply .github/workflows/slsa3_builder.yml:473]: Issue created https://github.com/slsa-framework/slsa-github-generator-go/issues/71

**Pull request merged**